ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/configuration/LedgerClientConfiguration.scala

```diff
@@ -3,18 +3,32 @@
 
 package com.daml.ledger.client.configuration
 
+import com.daml.ledger.client.configuration.LedgerClientConfiguration._
 import io.netty.handler.ssl.SslContext
 
 /**
-  * @param applicationId The string that will be used as an application identifier when issuing commands and retrieving transactions
-  * @param ledgerIdRequirement A [[LedgerIdRequirement]] specifying how the ledger identifier must be checked against the one returned by the LedgerIdentityService
-  * @param commandClient The [[CommandClientConfiguration]] that defines how the command client should be setup with regards to timeouts, commands in flight and command TTL
-  * @param sslContext If defined, the context will be passed on to the underlying gRPC code to ensure the communication channel is secured by TLS
-  * @param token If defined, the access token that will be passed by default, unless overridden in individual calls (mostly useful for short-lived applications)
+  * @param applicationId         The string that will be used as an application identifier when issuing commands and retrieving transactions
+  * @param ledgerIdRequirement   A [[LedgerIdRequirement]] specifying how the ledger identifier must be checked against the one returned by the LedgerIdentityService
+  * @param commandClient         The [[CommandClientConfiguration]] that defines how the command client should be setup with regards to timeouts, commands in flight and command TTL
+  * @param sslContext            If defined, the context will be passed on to the underlying gRPC code to ensure the communication channel is secured by TLS
+  * @param token                 If defined, the access token that will be passed by default, unless overridden in individual calls (mostly useful for short-lived applications)
+  * @param maxInboundMessageSize The maximum size of the response headers.
   */
 final case class LedgerClientConfiguration(
     applicationId: String,
     ledgerIdRequirement: LedgerIdRequirement,
     commandClient: CommandClientConfiguration,
     sslContext: Option[SslContext],
-    token: Option[String] = None)
+    token: Option[String] = None,
+    maxInboundMessageSize: Int = DefaultMaxInboundMessageSize,
+)
+
+object LedgerClientConfiguration {
+
+  // This needs to be large because we often include parts of the request in GRPC errors, which are
+  // returned in the response headers. A large request could lead to the client choking on the
+  // response headers if the limit is too low.
+  // The default limit set by Netty is 8 KB, which often leads to errors.
+  val DefaultMaxInboundMessageSize: Int = 1024 * 1024 // 1 MB
```

Comment:
Seems a bit high, do we really want to allow dumps so large? I think that the built-in limit is fine, as the dump is intended mostly for debugging purposes when developing small things. For larger dumps the process of inspecting the textual dump likely makes little sense.

Reply:
DO you think we should leave it to 8 KB then, and just truncate?

Reply:
I think 8KB is fine. At that point, the dump is already basically unreadable and there should exist some other way of dealing with it.

Reply:
yeah, I think the error message should describe the error well. We can log the full transaction in DEBUG mode on the server side.
Depending on where the response header limit error is thrown, maybe we can install a generic interceptor that does the truncation at a safe length (a few hundred chars should be enough). This is to ensure that the user always gets to see the real error.

Reply:
That's exactly what the [`TruncatedStatusInterceptor`](https://github.com/digital-asset/daml/pull/6807/files#diff-969b0fe6e367db5e9d13cc1601f181d8) does. 🙂

Reply:
Reduced back to 8 KB (the default), and error messages are now truncated to 4 KB.